bindings/python/iota_sdk/secret_manager/secret_manager.py

```diff
@@ -0,0 +1,177 @@
+# Copyright 2023 IOTA Stiftung
+# SPDX-License-Identifier: Apache-2.0
+
+from iota_sdk import create_secret_manager, call_secret_manager_method
+from json import dumps, loads
+import humps
+
+
+class LedgerNanoSecretManager(dict):
+    """Secret manager that uses a Ledger Nano hardware wallet or Speculos simulator.
+    """
+
+    def __init__(self, is_simulator):
+        """Initialize a ledger nano secret manager.
+        """
+
+        dict.__init__(self, ledgerNano=is_simulator)
+
+
+class MnemonicSecretManager(dict):
+    """Secret manager that uses a mnemonic in plain memory. It's not recommended for production use. Use LedgerNano or Stronghold instead.
+    """
+
+    def __init__(self, mnemonic):
+        """Initialize a mnemonic secret manager.
+        """
+
+        dict.__init__(self, mnemonic=mnemonic)
+
+
+class SeedSecretManager(dict):
+    def __init__(self, seed):
+        """Initialize a seed secret manager.
+        """
+
+        dict.__init__(self, hexSeed=seed)
+
+
+class StrongholdSecretManager(dict):
+    """Secret manager that uses Stronghold.
+    """
+
+    def __init__(self, snapshot_path, password):
+        """Initialize a stronghold secret manager.
+        """
+
+        dict.__init__(self, stronghold=StrongholdSecretManager.Inner(
+            snapshot_path, password))
+
+    class Inner(dict):
+        def __init__(self, snapshot_path, password):
+            dict.__init__(self, password=password, snapshotPath=snapshot_path)
+
+
+class SecretManagerError(Exception):
+    """secret manager error"""
+    pass
+
+
+class SecretManager():
+    def __init__(self, secret_manager):
+        self.handle = create_secret_manager(dumps(secret_manager))
+
+    def _call_method(self, name, data=None):
+        """Dumps json string and call call_secret_manager_method()
+        """
+        message = dumps({
+            'name': name,
+            'data': data
```

Comment:
In the client we first check if data is not None, shouldn't we do the same?

Reply:
Yes updated